src/core/QueryManager.ts

```
@@ -384,25 +384,43 @@ export class QueryManager {
           }
         } else {
           try {
-            const resultFromStore: ApolloQueryResult<T> = {
-              data: readQueryFromStore<T>({
-                store: this.getDataWithOptimisticResults(),
-                query: this.queryDocuments[queryId],
-                variables: queryStoreValue.previousVariables || queryStoreValue.variables,
-                returnPartialData: options.returnPartialData || noFetch,
-                config: this.reducerConfig,
-                previousResult: lastResult && lastResult.data,
-              }),
-              loading: isNetworkRequestInFlight(queryStoreValue.networkStatus),
-              networkStatus: queryStoreValue.networkStatus,
-            };
+            const { result: data, isMissing } = diffQueryAgainstStore({
+              store: this.getDataWithOptimisticResults(),
+              query: this.queryDocuments[queryId],
+              variables: queryStoreValue.previousVariables || queryStoreValue.variables,
+              returnPartialData: true,
+              config: this.reducerConfig,
+              previousResult: lastResult && lastResult.data,
+            });
+
+            let resultFromStore: ApolloQueryResult<T>;
+
+            // If there is some data missing and the user has told us that they
+            // do not tolerate partial data then we want to return the previous
+            // result and mark it as stale.
+            if (isMissing && !(options.returnPartialData || noFetch)) {
+              resultFromStore = {
+                data: lastResult.data,
```

In this case we should throw an error then, correct?